### DISCUSSION TITLE
Laravel Adjacency List / CTE Support

### DISCUSSION THREAD

**Discussion by ahmadmmughal** [planetscale/discussion#499] [Bug Reports] [answered]
I'm using a Laravel package `laravel-adjacency-list` whose query seems to be breaking when run on Planetscale. Digging deeper seems to point to CTE support not being available on Planetscale. But I just want to confirm, since Vitess seems to have had support for it since v13 as well as I have seem some video lessons on CTEs on Planetscale itself, leading me to believe that maybe the support is there hopefully and it's the package that needs adjustments.

**Comment by orware** [comment, marked as answer]
Hello @ahmadmmughal,

As far as I'm aware this support is still pending within Vitess as detailed in https://github.com/vitessio/vitess/issues/4099 and in the [Vitess documentation itself](https://vitess.io/docs/17.0/reference/compatibility/mysql-compatibility/#window-functions-and-ctes).

If you have a link to the support you were mentioning from v13 that you can share though I wouldn't mind reading through that information.

Our MySQL course, if that's where you saw the videos that mentioned CTEs, covers MySQL 8 generally so it may use some of the features that Vitess may still have compatibility issues with as detailed over on the following page:
https://planetscale.com/docs/reference/mysql-compatibility#mysql-compatibility-limitations

**Reply by ahmadmmughal** [reply]
Yes indeed you are right. It was my lack of understanding around Vitess' supported features, apologies.
